### ISSUE TITLE
Thread parked forever 

### ISSUE THREAD

**Issue by jilen** [eclipse/openj9#12377]
Java -version output
--------------------
```
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment AdoptOpenJDK (build 16+36)
Eclipse OpenJ9 VM AdoptOpenJDK (build openj9-0.25.0, JRE 16 Linux amd64-64-Bit Compressed References 20210311_69 (JIT enabled, AOT enabled)
OpenJ9   - 022d65424
OMR      - 09514431e
JCL      - 0c11227a21a based on jdk-16+36)
```

Summary of problem
------------------
This happens while app is waiting for  initializing, and other threads in the pool is waiting for this thread.
And then the initializing thread (application-akka.actor.default-dispatcher-12) is parked forever and never complete.
```
"application-akka.actor.default-dispatcher-12" prio=5 Id=52 WAITING
	at java.base@16/jdk.internal.misc.Unsafe.park(Native Method)
	at java.base@16/java.util.concurrent.locks.LockSupport.park(LockSupport.java:211)
	at java.base@16/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:714)
	at java.base@16/java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1046)
	at cats.effect.internals.IOPlatform$.$anonfun$unsafeResync$2(IOPlatform.scala:52)
	at cats.effect.internals.IOPlatform$$$Lambda$527/0x0000000000000000.apply$mcV$sp(Unknown Source)
	at app//scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at akka.dispatch.MonitorableThreadFactory$AkkaForkJoinWorkerThread$$anon$3.block(ThreadPoolBuilder.scala:174)
	at java.base@16/java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3127)
	at akka.dispatch.MonitorableThreadFactory$AkkaForkJoinWorkerThread.blockOn(ThreadPoolBuilder.scala:172)
	at akka.dispatch.BatchingExecutor$BlockableBatch.blockOn(BatchingExecutor.scala:109)
	at scala.concurrent.package$.blocking(package.scala:124)
	at cats.effect.internals.IOPlatform$.unsafeResync(IOPlatform.scala:52)
	at app//cats.effect.IO.unsafeRunTimed(IO.scala:342)
	at app//cats.effect.IO.unsafeRunSync(IO.scala:256)
```
```
"application-akka.actor.default-dispatcher-6" prio=5 Id=27 BLOCKED on zd.micro.app.server.play.AppComponents@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-7" prio=5 Id=36 BLOCKED on zd.micro.app.server.play.AppComponents@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-8" prio=5 Id=37 BLOCKED on zd.micro.app.server.play.AppComponents@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-9" prio=5 Id=38 BLOCKED on zd.micro.app.server.play.AppComponents@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-10" prio=5 Id=50 BLOCKED on zd.micro.app.server.play.AppComponents@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-11" prio=5 Id=51 BLOCKED on zd.micro.app.server.play.AppComponents@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
"application-akka.actor.default-dispatcher-12" prio=5 Id=52 WAITING
"application-akka.actor.default-dispatcher-13" prio=5 Id=53 BLOCKED on zd.micro.app.server.play.AppComponents@9bc01565 owned by "application-akka.actor.default-dispatcher-12" Id=52
```

Switch to hotspot jdk gets things fixed.

**Comment by keithc-ca** [comment]
Can you provide instructions on how to reproduce this problem?

**Comment by pshipton** [comment]
It may be worth trying the milestone 2 build for the next release in case the problem has been fixed.
https://github.com/AdoptOpenJDK/openjdk16-binaries/releases/tag/jdk-16%2B36_openj9-0.26.0-m2

i.e. https://github.com/AdoptOpenJDK/openjdk16-binaries/releases/download/jdk-16%2B36_openj9-0.26.0-m2/OpenJDK16U-jdk_x64_linux_openj9_16_36_openj9-0.26.0-m2.tar.gz

**Comment by jilen** [comment]
Might caused something by app it self,  openjdk 16 have also suffer this, but seems very rarely compared to openj9
